projects/dsp-ui/src/lib/viewer/views/property-view/property-view.component.ts

```diff
@@ -99,4 +97,24 @@ export class PropertyViewComponent implements OnInit, OnDestroy {
         this.addButtonIsVisible = true;
         this.propID = undefined;
     }
+
+    /**
+     * Given a resource property, check if an add button should be displayed under the property values
+     *
+     * @param prop the resource property
+     */
+    addValueIsAllowed(prop: PropertyInfoValues): boolean {
+        // check if cardinality allows for a value to be added
+        if (prop.guiDef.cardinality !== 1 ||
```

Comment:
@tobiasschweizer How can I get an instance of a `ResourceClassDefinition` to use for `CardinalityUtil.createdValueForPropertyAllowed`? I think that's what I'd want to use instead of this logic.

Reply:
the definition should be contained in the `ReadResource`'s `entityInfo.classes`

Reply:
`entityInfo.classes` has a type of `ResourceClassDefinitionWithPropertyDefinition` which is an extension of `ResourceClassDefinition` but I'm getting an error when I try to pass it in.

Reply:
there are some methods to get only certain defs

Reply:
but I think these work only for property defs

Reply:
What kind of error do you get? Can you use type assertions to make it work?

Otherwise we can look at tit on Monday.

Reply:
<img width="1620" alt="Screen Shot 2020-09-18 at 15 51 33" src="https://user-images.githubusercontent.com/60604010/93605310-dde78580-f9c6-11ea-817f-53fd90e97a70.png">

We can look at it on Monday; here's the error message. Type assertion doesn't seem to work either and it suggests to assert it as type unknown first, which also didn't help.

Reply:
I will try this myself and get back to you later this morning.

Reply:
@mdelez Could you try this `CardinalityUtil.createValueForPropertyAllowed(prop.propDef.id, prop.values.length, this.parentResource.entityInfo.classes[this.parentResource.type]);`

Reply:
That was the missing part :) Thanks!!